apps/ui/src/composables/useActions.ts

```diff
@@ -14,6 +14,8 @@ import type {
 } from '@/types';
 import type { Connector, StrategyConfig } from '@/networks/types';
 
+const offchainNetworkId = offchainNetworks.filter(network => enabledNetworks.includes(network))[0];
```

Comment:
I'd rather not have hardcoded network there, and instead use the network of the space, and in actions for each network `throw new Error("Not supported)`.

This way we don't complicate this composable (no need for hardcoding there), but also will support new networks automatically (and both s/s-tn will work at the same time).

Reply:
We use sequencer from v1 to handle all follows, offchain and onchain

Reply:
This doesn't work though, does it? I tried following my space on `sep` and it failed with `unknown space` error.

I still kind of feel like if all networks use the same action it should still be encapsulated in `actions` if `starknet` network uses `OffchainClient` then we should let it use it. We shouldn't leak internal implementation details outside of actions, even at the cost of slight repetition.

Reply:
You can only follow testnet from testnet env

We can test your approach

Reply:
The we should also move loadfollows inside each network, even though it can be done only once

Reply:
How do we expect the final version to work that supports onchain spaces as well? Would we have single query to load both onchain and offchain spaces or there will need to be separate requests?

If there would be separate then I think it would actually make sense to have separate `loadFollows` calls, but if it's single call then it's a bit unfortunate. Maybe we can think about it later. Could you add comment in the code there to mention that it maybe should be per-network instead of using hardcoded network with link tho this comment chain?

Reply:
The final version will come with this PR https://github.com/snapshot-labs/sx-monorepo/pull/294/files#diff-8bc937315cec4024222313b0e13597dfa00281e562d829047335da6b1dd811a4

This current PR already support onchain spaces, but unless you're on testnet, you can only follow mainnet onchain spaces. (your local with `yarn dev` can only follow `eth` spaces)

The next PR will add additional functions to the network api such as `loadAlias` and `setAlias`, which load and set the alias from the hub/sequencer. As said in the pitch, we will use sequencer from v1 to handler all alias, onchain and offchain. 

If we want to segregate things properly, each network should implement the `loadAlias`, `loadFollows` functions, which connect to the same API, and do the same thing. 

Same thing for `follows` and `unfollow`, on the sx.js side.

`loadFollows` is a single request, and fetch all follows from all networks

Reply:
> I'd rather not have hardcoded network there, and instead use the network of the space, and in actions for each network `throw new Error("Not supported)`.
> 
> This way we don't complicate this composable (no need for hardcoding there), but also will support new networks automatically (and both s/s-tn will work at the same time).

`s` and `s-tn` can not work at the same time, sequencer will accept only one or the other

Reply:
> If we want to segregate things properly, each network should implement the loadAlias, loadFollows functions, which connect to the same API, and do the same thing.
Same thing for follows and unfollow, on the sx.js side.
loadFollows is a single request, and fetch all follows from all networks

With this context I'd say that maybe follows/aliases shouldn't be part of networks at all, as it's not network specific. But we can keep it for now I guess. Can you create an issue for it, to discuss how should it implemented? (specifically if we should remove non-network specific functionality out of networks)?

Reply:
> This current PR already support onchain spaces, but unless you're on testnet, you can only follow mainnet onchain spaces. (your local with yarn dev can only follow eth spaces)

Do we know when mainnet sequencer will support offchain spaces?

Reply:
> > This current PR already support onchain spaces, but unless you're on testnet, you can only follow mainnet onchain spaces. (your local with yarn dev can only follow eth spaces)
> 
> Do we know when mainnet sequencer will support offchain spaces?

Sequencer already support both onchain and offchain, we just can not mix mainnet and testnet.

Reply:
> > If we want to segregate things properly, each network should implement the loadAlias, loadFollows functions, which connect to the same API, and do the same thing.
> > Same thing for follows and unfollow, on the sx.js side.
> > loadFollows is a single request, and fetch all follows from all networks
> 
> With this context I'd say that maybe follows/aliases shouldn't be part of networks at all, as it's not network specific. But we can keep it for now I guess. Can you create an issue for it, to discuss how should it implemented? (specifically if we should remove non-network specific functionality out of networks)?

This is true. I guess this will also apply to the upcoming user profile feature, which also will require some stuff like profile update (bio, avatar, etc...), and are not network specific

Reply:
Issue opened: https://github.com/snapshot-labs/sx-monorepo/issues/359